lit_gpt/utils.py

```diff
@@ -497,14 +497,14 @@ def map_old_state_dict_weights(state_dict: Dict, mapping: Mapping, prefix: str)
 
 
 def get_default_supported_precision(training: bool) -> str:
-    """Return default precision that is supported by the hardware.
+    """Return default precision that is supported by the hardware: either `bf16` or `16`.
 
     Args:
         training: `-mixed` or `-true` version of the precision to use
 
     Returns:
         default precision that is suitable for the task and is supported by the hardware
     """
-    if not torch.cuda.is_available() or torch.cuda.is_bf16_supported():
-        return "bf16-mixed" if training else "bf16-true"
-    return "16-mixed" if training else "16-true"
+    if torch.backends.mps.is_available() or (torch.cuda.is_available() and not torch.cuda.is_bf16_supported()):
```

Comment:
Can you check MPS availability by using this staticmethod? https://github.com/Lightning-AI/lightning/blob/master/src/lightning/fabric/accelerators/mps.py#L65

Reply:
Not sure why it's necessary. That static method checks that the `mps` device can be supported by Fabric, while here I check that the precision is supported, and all `mps` devices doesn't support `bf16`. Maybe with Apple M3 it will be changed, but for now it is so.
Or you want it to be aligned with the code from Fabric?

Reply:
Yes, I wanted it to be aligned to cover any edge cases that are already considered by the MPSAccelerator.is_available implementation